### ISSUE TITLE
Add easier way to scale keyframes in AnimationPlayer Editor

### ISSUE THREAD

**Issue by mrkdji** [godotengine/godot-proposals#3532]
### Describe the project you are working on

I'm making a cutscene with the AnimationPlayer.

### Describe the problem or limitation you are having in your project

The current way to scale keyframes in AnimationPlayer editor is cumbersome

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Scale a selection of animation player keyframes by clicking Left Mouse Button on one of the selection extremities and eventually by holding a mouse modifier (for instance ALT).

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

It would be nice if the AnimationPlayer editor keyframes could be scaled like the notes in REAPER's MIDI editor

![scale_notes](https://user-images.githubusercontent.com/38851942/141336118-56d5bb30-3c4f-471d-9963-0206f295e24e.gif)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I'm not sure.

### Is there a reason why this should be core and not an add-on in the asset library?

AnimationPlayer editor is core.

**Comment by vyshliy** [comment]
I completely agree. This feature will greatly speed up the creation of animations. Same as mirroring selected frames.

**Comment by Calinou** [comment]
@vyshliy Please don't bump issues without contributing significant new information. Use the :+1: reaction button on the first post instead.

**Comment by vyshliy** [comment]
Blender also has a great implementation of this feature. It speeds up animation editing at times. 

https://user-images.githubusercontent.com/72736632/150919428-6d94435d-7bc7-4e4b-bf24-b0b9479535e1.mp4

**Comment by vyshliy** [comment]
To be more specific. Blender has a convenient implementation of the proposed feature in terms of usability. 

https://user-images.githubusercontent.com/72736632/150934711-acd9cf6e-64d4-4dbb-b3bd-4d6bba9b2486.mp4

- Blender uses the mouse modifier "S" to scale frames in the animation tab that match the scale modifier in the other tabs. 
- Selected frames are scaled relative to an active frame allowing to scale from the desired position in the timeline, so there is no need to move the selected frames afterwards. 
- Negative scaling allows to mirror frames without any additional clicks.
- Blender also allows to scale curve values together with the frame position, which may be related to #3141 

https://user-images.githubusercontent.com/72736632/150934741-5f444be6-14f7-410e-a33d-91b654b16570.mp4

**Comment by hsandt** [comment]
Unity also has a nice way to represent this with vertical blue bars around the selected frames:

![image](https://github.com/godotengine/godot-proposals/assets/1067772/2b6bb079-f73d-4ffc-8417-1f1a9b2583ca)

Source: https://docs.unity3d.com/Manual/animeditor-AdvancedKeySelectionAndManipulation.html

It is also used to move groups of keys (which Godot already supports, but instead of starting mouse drag from a specific key, you can start with the mouse cursor anywhere inside the selection box).

Note the final section about _Ripple editing_, a feature that allows to push non-selected keys while performing the Move or Scale operation. Certainly something we'll have to think of when implementing the Scale (and why not also Move, so we improve the existing one).

**Comment by hsandt** [comment]
Just saw this other proposal to extend animation duration via mouse drag (https://github.com/godotengine/godot-proposals/issues/10030) and it occurred to me that we may also want to allow stretching keyframes beyond the max duration, automatically extending the animation duration accordingly.

**Comment by phantomdesvin** [comment]
As an animator I'm personally used to how it works in most animation programs I've work with, where you just select some keyframes, hold ALT and LeftClick on the first or the last keyframe and drag. The point of reference is the other extreme of the selection (if you clicked on the last keyframe of the selection, the reference would be the first keyframe of the selection, so that first keyframe would mantain it's position in the timeline). In practice, we (animators) never scale from any point other than the start or the end of the selection because it would be a bad practice to risk interfering with previous and next keyframes at the same time and I can't see a reason to do that, so I wouldn't sacrifice the simplicity of this common workflow to accomodate that option, that can be achieved just scaling from one extreme first and then dragging the selection to the side.

Moving/scaling keyframes beyond the max duration of the animation is common practice in animation programs, but automatically extending the animation duration accordingly is not, and I can see that being undesirable in some cases and, if that other proposal about time is implemented, it should already be quick and easy to modify when actually needed.

**Comment by PhairZ** [comment]
I've made a [PR](https://github.com/godotengine/godot/pull/105151) recently closing this issue. No one has reviewed it so far, so if anyone's open for testing, please go ahead.

**Comment by GourabSN** [comment]
I also need this feature in Godot but it is not implemented yet. So I made a simple addon for it [Godot Keyframe Scaling](https://github.com/GourabSN/Keyframe-Scaling-GodotAddon.git). 
I wish Godot could do it like Blender does. The [Addon](https://github.com/GourabSN/Keyframe-Scaling-GodotAddon.git) does the work, but it would be more intuitive if could be done as in Blender.

This is how Blender does it:

![Image](https://github.com/user-attachments/assets/bbce90f4-0d15-4321-ba93-84d0216dbbd9)

This is how my [addon](https://github.com/GourabSN/Keyframe-Scaling-GodotAddon.git) does it:

![Image](https://github.com/user-attachments/assets/6ed5ef9f-79b2-47e0-87e6-b1fbb6f85912)
